src/extension.ts

```diff
@@ -99,10 +132,27 @@ class Licenser {
             vscode.window.showErrorMessage("No directory is opened.");
             return;
         }
+
+        this._chooseLicenseType().then(licenseType => {
+            if (!isNullOrUndefined(licenseType)) {
+                const license = this.getLicense(licenseType);
+                this._doCreateLicense(root, license);
+            }
+        });
+    }
+
+    private _chooseLicenseType(): Thenable<string> {
         let licenserSetting = vscode.workspace.getConfiguration("licenser");
         let licenseType = licenserSetting.get<string>("license");
-        const license = this.getLicense(licenseType);
 
+        if (isNullOrUndefined(licenseType) || licenseType.toLowerCase() == "choose from list") {
```

Comment:
Can you make the string "choose from list" as `const` value in this module or class just in case?

Reply:
Done!